swift-sdk/Internal/RequestCreator.swift

```diff
@@ -137,36 +138,33 @@ struct RequestCreator {
         }
         
         var itemsToSerialize = [[AnyHashable: Any]]()
+        
         for item in items {
             itemsToSerialize.append(item.toDictionary())
         }
         
-        var apiUserDict = [AnyHashable: Any]()
+        var apiUserDict: [AnyHashable: Any] = [:]
+        
         apiUserDict.setValue(for: keyValueForCurrentUser.key, value: keyValueForCurrentUser.value)
         
-        let body: [String: Any]
+        var body: [String: Any] = [JsonKey.Commerce.user: apiUserDict,
+                                   JsonKey.Commerce.items: itemsToSerialize,
+                                   JsonKey.Commerce.total: total]
+        
         if let dataFields = dataFields {
-            body = [JsonKey.Commerce.user: apiUserDict,
-                    JsonKey.Commerce.items: itemsToSerialize,
-                    JsonKey.Commerce.total: total,
-                    JsonKey.dataFields.jsonKey: dataFields]
-        } else {
-            body = [JsonKey.Commerce.user: apiUserDict,
-                    JsonKey.Commerce.items: itemsToSerialize,
-                    JsonKey.Commerce.total: total]
+            body[JsonKey.dataFields.jsonKey] = dataFields
         }
         
         return .success(.post(createPostRequest(path: Const.Path.trackPurchase, body: body)))
     }
     
     func createTrackPushOpenRequest(_ campaignId: NSNumber, templateId: NSNumber?, messageId: String?, appAlreadyRunning: Bool, dataFields: [AnyHashable: Any]?) -> Result<IterableRequest, IterableError> {
-        var body = [AnyHashable: Any]()
+        var body: [AnyHashable: Any] = [:]
```

Comment:
We should not have to declare the type that can be inferred. The original code is more concise and recommended.